tests/test_gflow.py

```diff
@@ -601,12 +601,12 @@ def test_rand_graph_pauliflow(self, fx_bg: PCG64, jumps: int) -> None:
             assert valid
 
     def test_corrections_from_pattern(self) -> None:
-        pattern = Pattern(input_nodes=range(5))
+        pattern = Pattern(input_nodes=list(range(5)))
```

Comment:
We shouldn't need to do that, as `Pattern` already constructs a list from `input_nodes`. The type annotation in `Pattern` could be relaxed to `Sequence` (or `Iterable`, though in that case, we should be cautious about transient iterables by using `self.__input_nodes` instead of `input_nodes` when calculating `self.__n_node` and `self.__output_nodes`).

Reply:
Let me work on this improvement in another PR for `Pattern`.